### PR TITLE
[ENHANCEMENT] adding Bitrise workflow for build failure Slack notification [skip ci]

### PR DESCRIPTION
**Pull request by NicolasMassart**
**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://docs.google.com/document/d/1VJLwTRsUw_5EDq_o8d6sSbXUAYENLurkRitYO45eY5o/edit?usp=sharing)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add `needs-qa` label when dev review is completed
5. Add `QA Passed` label when QA has signed off

## Description

_Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions,_
_1. What is the reason for the change?_
- Build have a success notification but nothing when it fails 

_2. What is the improvement/solution?_
- Add a new workflow for slack notification that runs after the build only in case of failure
- Make Slack notifications consistent (not par of the issue initially but otherwise not being consistent increases the complexity)
- Prevent `deploy-to-bitrise-io` steps to run even when everything failed before. Only notification should run after a failure.

This changes leverage the use of the Slack step and I included making the previously added notifications consistent.

It make the PR a bit large scoped but if we split into multiple PRs, I tend to think it will increase the complexity as the yml will be in a non consistent status until we consolidate.
Given it’s very clearly related and similar, I decided to have it all in this PR.
I leave the `release_to_stores` workflow announcement out of the scope for now here but it is a good candidate to move to the Slack step too, making everything even more consistent.

## NOTE

[skip ci] added to prevent auto trigger of the workflow on push and send notifications on actual Slack channel for no reason...
So testing with manual trigger and custom Slack creds env vars

## Screenshots

_If applicable, add screenshots and/or recordings to visualize the before and after of your change_

Failure notification will look like this:

image

(screenshot from the test slack sent from [failed on purpose build `#5918`](https://app.bitrise.io/build/3c05c77d-a366-4fc4-b261-27eb3b75a2ef))

## Issue

fixes MetaMask/mobile-planning#441

## Checklist

* [x] There is a related GitHub issue
* [x] Tests are included if applicable
* [x] Any added code is fully documented

### CODE REVIEW

File: bitrise.yml [hunk 1]
```diff
@@ -145,31 +237,6 @@ workflows:
           inputs:
             - abort_on_fail: 'yes'
             - access_token: $BITRISE_START_BUILD_ACCESS_TOKEN
-      - script@1:
```
Comment (NicolasMassart) [inline bitrise.yml:148]: Note, this echo script removed because it's useless now that we have notification.

File: bitrise.yml [hunk 2]
```diff
@@ -376,7 +448,7 @@ workflows:
                 cp -r $BITRISE_SOURCE_DIR/wdio/reports/junit-results/ $BITRISE_TEST_RESULT_DIR/
       - deploy-to-bitrise-io@1:
           inputs:
-              - deploy_path: $BITRISE_TEST_RESULT_DIR
+            - deploy_path: $BITRISE_TEST_RESULT_DIR
```
Comment (NicolasMassart) [inline bitrise.yml:451]: this is just whitespaces formatting fix

File: bitrise.yml [hunk 3]
```diff
@@ -54,6 +53,91 @@ workflows:
             - cache_local_deps: 'yes'
             - command: lint
           is_always_run: true
+
+  # Notifications utility workflows
+  # Provides values for commit or branch message and path depending on commit env setup initialised or not
+  _get_workflow_info:
```
Comment (sethkfman) [inline bitrise.yml:59]: I like these utility workflows. As this file grows, I think we may want to consider pushing these utility workflows into their own files.

**Review by sethkfman — APPROVED** [review]
LGTM

**Comment by DigitalsupportTeam** [comment]
We'd be happy to look into this for you
kindly file a support ticket below 
https://forms.gle/Wmpm6NLzYm9uXYuU9

File: bitrise.yml (hunk 3 above)
Comment (NicolasMassart) [reply bitrise.yml:59]: I'm not sure we can do that...

**Pull request merged**